app/views/conversations/index.html.erb

```diff
@@ -0,0 +1,36 @@
+<div class="box wide">
+  <h2 class="legend">Conversations</h2>
+  <section class="data zebra conversations">
+    <header>
+      <span class="partner">Partner</span>
+      <span class="received">Received</span>
+      <span class="subject">Subject</span>
+    </header>
+    <% @conversations.each do |conversation| %>
+      <%=
+      link_to(
+        conversation_path(conversation),
+        class: "conversation #{conversation_status_classes(conversation)}"
+      ) do
+      %>
+        <span class="partner">
+          <%= conversation.partner(of: @user).username %>
+        </span>
+        <span class="received"><%= time_ago_in_words_label(conversation.last_message.created_at) %></span>
+        <span class="subject"><%= conversation.subject %></span>
```

Comment:
Please include `conversation.messages.count` somewhere in this table (which is a a 1 + n query, so needs to be preloaded).

Reply:
Done. I added a `preload(:messages)` but after doing some quick perf tests (just looking at loading times in the log) I didn't see a difference between preloading and not. I also didn't get any kind of warning from bullet. If you know something I don't, I'll happily put a preload in there 🙂

Reply:
If you don't see a `select count...` query in the output of your `rails server` for every conversation, and bullet isn't complaining there, you're fine.

(Though maybe the query output I'm used to seeing comes from my `~/.pryrc`?)

```
require 'logger'
if Object.const_defined? 'Rails'
  Rails.logger = Logger.new STDOUT
end
# print SQL to STDOUT
if Object.const_defined? 'ActiveRecord'
  ActiveRecord::Base.logger = Logger.new STDOUT
  ActiveRecord::Base.clear_reloadable_connections!
end
```

Reply:
Grabbed your pryrc No bullet, but I do see the count for every query.

```sql
SELECT COUNT(*) FROM `messages` WHERE `messages`.`conversation_id` = 16
```

I just can't get a performance difference between preloading and not with ~100 conversations for a user. If there's another thing I should test, let's do it!

Reply:
I wouldn't expect any performance difference in your local dev, but it'll cause problems in prod because of the sheer number of queries all kicking off at once while there's other traffic on the database. I'll take a look on Wednesday on how to preload the count properly; I always have to tinker to get the right syntax.

Reply:
I looked it up. You'll need to add a `messages_cache` column to `Conversations` and [`counter_cache` to the `belongs_to`](https://guides.rubyonrails.org/v5.2/association_basics.html#options-for-belongs-to-counter-cache), then Rails will keep the count up-to-date without doing a `count(*)` query for every conversation.